collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -113,13 +112,14 @@
         TextView titleView = findViewById(R.id.form_title);
         titleView.setText(viewModel.getFormTitle());
 
-        if (map == null) { // tests set their maps directly
-            Context context = getApplicationContext();
-            map = MapProvider.createMapFragment(context);
+        MapFragment mapToAdd = map; // tests will have a value for the map field
+
+        if (mapToAdd == null) {
+            mapToAdd = MapProvider.createMapFragment(getApplicationContext());
```

Comment:
Yeah I think we should use Dagger for this. It feels like we're adding machinery in to mimic what it would do for us here.

Reply:
Sigh, ok, will do. 😔

Reply:
I should have referred back to https://github.com/opendatakit/collect/pull/3552#discussion_r360364242 before mentioning Dagger. We can't inject the `MapFragment` for exactly the reason that this bug is related to -- we don't want to set the field until the asynchronous initialization has completed. Alternately, we could keep extra state to know whether the initialization has completed or not but I don't think that's very clean. So I went the route of injecting a `MapProvider` singleton. I hope you will find this reasonable, @seadowg.

Reply:
For the record and so that perhaps someone can push back on me, some things that upset me about Dagger/our Dagger setup:
- the magic. We end up with a lot of generated code that feels brittle.
- the mental overhead. I thought I could inject into `BaseGeoMapActivity` not realizing that I couldn't because it's abstract. I had to track down that I was getting a no-op inject method and find my way to https://github.com/google/dagger/issues/489 since there's no warning. It might just be a familiarity thing but I feel like I end up with some kind of irritation like this every time I'm using Dagger.
- we wax poetic about separation of concerns and then we have `AppDependencyModule` and `AppDependencyComponent` which feel like really ugly grab-bags of stuff. Part of that is that the app itself isn't currently divided up into modules.